server/routerlicious/packages/services-shared/src/runner.ts

```diff
@@ -110,6 +114,9 @@ export function runService<T extends IResources>(
 			process.exit(0);
 		},
 		async (error) => {
+			if (error.uncaughtException) {
```

Comment:
Where do we set the `uncaughtException` of the error object? (sorry if I missed it)

I see that in line 56 we listen for the "uncaughtException" event, but do we throw an error anywhere so that this line 117 can catch it, and if yes, is the `uncaughtException` property set?

Also, something related to my other comment: perhaps we could have a `restartReason` property instead, and set it appropriately on each case. That would be more versatile than using a dedicated property for this particular scenario.

Reply:
We set uncaughtException in runner.stop() method (I did those changes in my previous PR: https://github.com/microsoft/FluidFramework/commit/819b496122bfae56fb6e809109dfb16188230b0f#diff-f6124496c29a4dbf4505e6b73c733714d733d580d3b0e5945d2bfd88f65683fbR148) 

And yes, updating the property to be more generic.